runtime/src/transaction_batch.rs

```diff
@@ -1,21 +1,37 @@
 use {
     crate::bank::Bank, solana_sdk::transaction::Result,
-    solana_svm_transaction::svm_transaction::SVMTransaction, std::borrow::Cow,
+    solana_svm_transaction::svm_message::SVMMessage,
 };
 
+pub enum OwnedOrBorrowed<'a, T> {
+    Owned(Vec<T>),
+    Borrowed(&'a [T]),
+}
+
+impl<T> core::ops::Deref for OwnedOrBorrowed<'_, T> {
+    type Target = [T];
+
+    fn deref(&self) -> &Self::Target {
+        match self {
+            OwnedOrBorrowed::Owned(v) => v,
+            OwnedOrBorrowed::Borrowed(v) => v,
+        }
+    }
+}
+
 // Represents the results of trying to lock a set of accounts
-pub struct TransactionBatch<'a, 'b, Tx: SVMTransaction + Clone> {
+pub struct TransactionBatch<'a, 'b, Tx: SVMMessage> {
```

Comment:
so, strictly speaking, we could use this more relaxed `SVMMessage` at the time of #2836?

Reply:
Yeah could have definitely, just messed up.
We just need to be able to get the `account_locks`, no need to have signatures in order to unlock.